src/compressed_tensors/transform/utils/utils.py

```diff
@@ -27,17 +27,28 @@ def get_matrix_size(module: torch.nn.Module, location: TransformLocation) -> int
     :param location: location on module
     :return: size of matrix
     """
-    assert isinstance(module, torch.nn.Linear)
-    if location in ("input", TransformLocation.WEIGHT_INPUT):
-        return module.in_features
-    else:
-        return module.out_features
+    if isinstance(module, torch.nn.Linear):
+        if location in ("input", TransformLocation.WEIGHT_INPUT):
+            return module.in_features
+        else:
+            return module.out_features
+    elif isinstance(module, torch.nn.Embedding):
+        if location in ("input", TransformLocation.WEIGHT_INPUT):
+            return module.num_embeddings
+        else:
+            return module.embedding_dim
+
+    raise ValueError(
+        f"Unsupported module type {type(module)}, "
+        "should be either Linear or Embedding."
+    )
 
 
 def apply_transform_weight(
-    weight: torch.Tensor,
+    transform_weight: torch.Tensor,
     value: torch.Tensor,
     location: TransformLocation,
+    is_linear: bool = True,
```

Comment:
I'm not a fan of `is_linear`, since we're currently making the assumption that `not is_linear => is_embedding`. It also leads to a lot of if statements, whereas I'd like to keep this function as clean as possible.

Maybe we can rename this to `apply_transform_weight_linear` and do something like this?
```
def apply_transform_weight(..., module_type):
    if module_type is torch.nn.Linear:
        apply_transform_weight_linear(...)
    elif module_type is torch.nn.Embedding:
         if location in (WEIGHT_INPUT, WEIGHT_OUTPUT):
             apply_transform_weight_linear(..., weight=weight.T)
         else:
             apply_transform_weight_linear(...)
    else:
        raise NotImplementedError(...)
```

Reply:
marking this as resolved, the implementation now is cleaner